File: ios/easy_blue_printer/Sources/easy_blue_printer/plugin/domain/usecases/ConfigurePrinterUseCase.swift
import Foundation

public class ConfigurePrinterUseCase {
    private let repository: BluetoothRepository

    public init(repository: BluetoothRepository) {
        self.repository = repository
    }

    public func execute(paperWidth: Int, heatingTime: Int?) {
        repository.configurePrinter(paperWidth: paperWidth, heatingTime: heatingTime)
    }
}

File: android/src/main/kotlin/com/maktubcompany/easy_blue_printer/plugin/domain/usecase/ConfigurePrinterUseCase.kt
package com.maktubcompany.easy_blue_printer.plugin.domain.usecase

import com.maktubcompany.easy_blue_printer.plugin.domain.repository.BluetoothRepository

class ConfigurePrinterUseCase(private val repository: BluetoothRepository) {
    fun execute(paperWidth: Int, heatingTime: Int?) {
        repository.configurePrinter(paperWidth, heatingTime)
    }
}
